command/repo.go

```diff
@@ -63,6 +82,146 @@ func repoClone(cmd *cobra.Command, args []string) error {
 	return utils.PrepareCmd(cloneCmd).Run()
 }
 
+func repoCreate(cmd *cobra.Command, args []string) error {
+	projectDir, projectDirErr := git.ToplevelDir()
+
+	orgName := ""
+	teamSlug, err := cmd.Flags().GetString("team")
+	if err != nil {
+		return err
+	}
+
+	var name string
+	if len(args) > 0 {
+		name = args[0]
+		if strings.Contains(name, "/") {
+			newRepo := ghrepo.FromFullName(name)
+			orgName = newRepo.RepoOwner()
+			name = newRepo.RepoName()
+		}
+	} else {
+		if projectDirErr != nil {
+			return projectDirErr
+		}
+		name = path.Base(projectDir)
+	}
+
+	isPublic, err := cmd.Flags().GetBool("public")
+	if err != nil {
+		return err
+	}
+	hasIssuesEnabled, err := cmd.Flags().GetBool("enable-issues")
+	if err != nil {
+		return err
+	}
+	hasWikiEnabled, err := cmd.Flags().GetBool("enable-wiki")
+	if err != nil {
+		return err
+	}
+	description, err := cmd.Flags().GetString("description")
+	if err != nil {
+		return err
+	}
+	homepage, err := cmd.Flags().GetString("homepage")
+	if err != nil {
+		return err
+	}
+
+	// TODO: move this into constant within `api`
+	visibility := "PRIVATE"
+	if isPublic {
+		visibility = "PUBLIC"
+	}
+
+	input := api.RepoCreateInput{
+		Name:             name,
+		Visibility:       visibility,
+		OwnerID:          orgName,
+		TeamID:           teamSlug,
+		Description:      description,
+		Homepage:         homepage,
+		HasIssuesEnabled: hasIssuesEnabled,
+		HasWikiEnabled:   hasWikiEnabled,
+	}
+
+	ctx := contextForCommand(cmd)
+	client, err := apiClientForContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	repo, err := api.RepoCreate(client, input)
+	if err != nil {
+		return err
+	}
+
+	out := cmd.OutOrStdout()
+	greenCheck := utils.Green("✓")
+	isTTY := false
+	if outFile, isFile := out.(*os.File); isFile {
+		isTTY = utils.IsTerminal(outFile)
+		if isTTY {
+			// FIXME: duplicates colorableOut
+			out = utils.NewColorable(outFile)
+		}
+	}
+
+	if isTTY {
+		fmt.Fprintf(out, "%s Created repository %s on GitHub\n", greenCheck, ghrepo.FullName(repo))
+	} else {
+		fmt.Fprintln(out, repo.URL)
+	}
+
+	remoteURL := repo.URL + ".git"
+
+	if projectDirErr == nil {
+		// TODO: use git.AddRemote
+		remoteAdd := git.GitCommand("remote", "add", "origin", remoteURL)
+		remoteAdd.Stdout = os.Stdout
+		remoteAdd.Stderr = os.Stderr
+		err = utils.PrepareCmd(remoteAdd).Run()
+		if err != nil {
+			return err
+		}
+		if isTTY {
+			fmt.Fprintf(out, "%s Added remote %s\n", greenCheck, remoteURL)
+		}
+	} else if isTTY {
+		doSetup := false
+		// TODO: use overridable Confirm
```

Comment:
given that i added that in repo fork i'm ok with follow-up work increasing test coverage one all the repo stuff is in.

Reply:
Yes, that's what the TODO work is all about. Furthermore, I only activate prompts if stdout is a tty (come to think of it, we should also check if stdin is tty, otherwise prompts won't work), and currently we don't have a way of emulating a tty in tests.